File: debug-server.mjs
import { Client } from 'ssh2';

const conn = new Client();
function run(cmd) {
  return new Promise((resolve, reject) => {
    console.log(`[SSH] $ ${cmd}`);
    conn.exec(cmd, (err, stream) => {
      if (err) return reject(err);
      let out = '', err2 = '';
      stream.on('data', d => { out += d; process.stdout.write(d.toString()); });
      stream.stderr.on('data', d => { err2 += d; process.stderr.write(d.toString()); });
      stream.on('close', code => code === 0 ? resolve(out) : reject(new Error(`Exit ${code}: ${err2 || out}`)));
    });
  });
}

conn.on('ready', async () => {
  try {
    await run('ls -la /var/www/alshifa/server.js 2>/dev/null || echo "NO server.js"');
    await run('ls -la /var/www/alshifa/.next/ 2>/dev/null | head -15');
    await run('cat /var/www/alshifa/.next/BUILD_ID 2>/dev/null || echo "NO BUILD_ID"');
    await run('ls -la /var/www/alshifa/.next/server/ 2>/dev/null | head -10');
    await run('ls -la /var/www/alshifa/.next/static/ 2>/dev/null | head -5');
    await run('ls /var/www/alshifa/node_modules/next 2>/dev/null | head -5 || echo "NO next module"');
    await run('head -5 /var/www/alshifa/server.js 2>/dev/null');
  } catch (err) {
    console.error('Error:', err.message);
  } finally {
    conn.end();
  }
});
conn.on('error', err => { console.error('SSH Error:', err.message); process.exit(1); });
conn.connect({ host: '66.42.113.157', port: 22, username: 'root', password: '@Ky8hTV{]NgqV@p)', readyTimeout: 15000 });

File: debug_server.py
import paramiko
import time

HOST = "66.42.113.157"
USER = "root"
PASS = "@Ky8hTV{]NgqV@p)"

def run_cmd(ssh, cmd, timeout=30):
    stdin, stdout, stderr = ssh.exec_command(cmd, timeout=timeout)
    return stdout.read().decode(), stderr.read().decode()

def main():
    ssh = paramiko.SSHClient()
    ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
    ssh.connect(HOST, username=USER, password=PASS, timeout=30)

    # Check .env file
    out, err = run_cmd(ssh, "cat /var/www/alshifa/.env 2>&1")
    print(f".env file:\n{out}")

    # Try to run server.js directly to see the error
    out, err = run_cmd(ssh, "cd /var/www/alshifa && node server.js 2>&1", timeout=15)
    print(f"Direct run output:\n{out}")
    print(f"Direct run error:\n{err}")

    # Check if node_modules exists in standalone
    out, err = run_cmd(ssh, "ls /var/www/alshifa/.next/standalone/node_modules/ 2>&1 | head -10")
    print(f"Standalone node_modules: {out}")

    ssh.close()

if __name__ == "__main__":
    main()
